user_interface/EbikeUIController/can.cpp

```diff
@@ -0,0 +1,121 @@
+    can.onReceive(MB1, updateCAN_MotorVoltages_Struct); // Set mailbox 1 to receive system parameters
+    can.onReceive(MB2, updateCAN_RealCurrents_Struct); // Set mailbox 2 to receive system parameters
+    can.onReceive(MB3, updateCAN_CalcValues_Struct); // Set mailbox 3 to receive system parameters
+    can.onReceive(MB4, updateCAN_BmsSoc_Struct); // Set mailbox 4 to receive system parameters
+    can.setMB(MB5, TX); // Set mailbox 5 as transmit
+}
+
+//function used as callback when bike status message enters mailbox
+void updateCAN_BikeStatus_Struct(const CAN_message_t &msg) {
+    // parses CAN message values into variables  
+    CAN_BikeStatus_Struct.timestamp = (msg.buf[0] << 8) | msg.buf[1];
+    CAN_BikeStatus_Struct.speed = (msg.buf[2] << 8) | msg.buf[3];
+    CAN_BikeStatus_Struct.temp = (msg.buf[4] << 8) | msg.buf[5];
+    CAN_BikeStatus_Struct.throttle = msg.buf[6];
+    CAN_BikeStatus_Struct.error = msg.buf[7];
+    CAN_MB0_BikeStatus_Flag = true; // updated values have been read
+}
+
+//function used as callback when motor voltage message enters mailbox
+void updateCAN_MotorVoltages_Struct(const CAN_message_t &msg) {
+    // parses CAN message values into variables 
+    CAN_MotorVoltages_Struct.timestamp = (msg.buf[0] << 8) | msg.buf[1];
+    CAN_MotorVoltages_Struct.vDC = (msg.buf[2] << 8) | msg.buf[3];
+    CAN_MotorVoltages_Struct.vA = (msg.buf[4] << 8) | msg.buf[5];
+    CAN_MotorVoltages_Struct.vB = (msg.buf[6] << 8) | msg.buf[7];
+    CAN_MB1_MotorVoltages_Flag = true; // updated values have been read
+}
+
+//function used as callback when real current message enters mailbox
+void updateCAN_RealCurrents_Struct(const CAN_message_t &msg) {
+    // parses CAN message values into variables 
+    CAN_RealCurrents_Struct.timestamp = (msg.buf[0] << 8) | msg.buf[1];
+    CAN_RealCurrents_Struct.iDC = (msg.buf[2] << 8) | msg.buf[3];
+    CAN_RealCurrents_Struct.iA = (msg.buf[4] << 8) | msg.buf[5];
+    CAN_RealCurrents_Struct.iB = (msg.buf[6] << 8) | msg.buf[7];
+    CAN_MB2_RealCurrents_Flag = true; // updated values have been read
+}
+
+//function used as callback when calculated values message enters mailbox
+void updateCAN_CalcValues_Struct(const CAN_message_t &msg) {
+    // parses CAN message values into variables 
+    CAN_CalcValues_Struct.timestamp = (msg.buf[0] << 8) | msg.buf[1];
+    CAN_CalcValues_Struct.iqRef = (msg.buf[2] << 8) | msg.buf[3];
+    CAN_CalcValues_Struct.iqFdb = (msg.buf[4] << 8) | msg.buf[5];
+    CAN_CalcValues_Struct.power = (msg.buf[6] << 8) | msg.buf[7];
+    CAN_MB3_CalcValues_Flag = true; // updated values have been read
+}
+
+//function used as callback when BMS message enters mailbox
+void updateCAN_BmsSoc_Struct(const CAN_message_t &msg) {
+    // parses CAN message values into variables 
+    CAN_BmsSoc_Struct.soc = (msg.buf[0] << 8) | msg.buf[1];
+    CAN_BmsSoc_Struct.soh = (msg.buf[2] << 8) | msg.buf[3];
+    CAN_MB4_BmsSoc_Flag = true; // updated values have been read
+}
+
```

Comment:
In order to check if there is a new message, we should have functions that return a copy of a message flag. something like:
`bool canPendingBikeStatusMsg(){
    return CAN_MB0_BikeStatus_Flag;
}
`